### PR TITLE
CORE-14306. [KMTEST] Fix 3 Clang-Cl warnings about Status and Callbacks

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

JIRA issue: [CORE-14306](https://jira.reactos.org/browse/CORE-14306)

@gedmurphy

## Proposed changes

- 1 "warning: variable 'Status' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]"
Double-check this fix, as I am unsure which default value should be used.
- 2 "warning: comparison of unsigned expression < 0 is always false [-Wtautological-unsigned-zero-compare]"
- Also update license header.
Double-check this fix, as I assumed a year for Thomas, as I am unsure why he is listed here.

plus

- Use a consistent type for "i" and fix a comment
Mark, never look at this additional commit...

### CODE REVIEW

File: modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c [hunk 1]
```diff
@@ -146,7 +147,7 @@ DriverEntry(
     //__debugbreak();
     DPRINT("DriverEntry\n");
 
-    RtlZeroMemory(&FilterData, sizeof(FILTER_DATA));
+    RtlZeroMemory(&FilterData, sizeof(FilterData));
```
Comment (ThFabba) [inline modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: UNRELATED CHANGE

File: modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c [hunk 2]
```diff
@@ -182,7 +183,7 @@ DriverEntry(
 
     /* call TestEntry */
     RtlInitUnicodeString(&DeviceName, DeviceNameBuffer);
-    DeviceName.MaximumLength = sizeof DeviceNameBuffer;
+    DeviceName.MaximumLength = sizeof(DeviceNameBuffer);
```
Comment (ThFabba) [inline modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: UNRELATED CHANGE

File: modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c [hunk 3]
```diff
@@ -514,12 +518,12 @@ FilterPreOperation(
 {
     FLT_PREOP_CALLBACK_STATUS Status;
     UCHAR MajorFunction;
-    INT i;
+    ULONG i;
 
     Status = FLT_PREOP_SUCCESS_NO_CALLBACK;
     MajorFunction = Data->Iopb->MajorFunction;
 
-    for (i = 0; i < sizeof(Callbacks) / sizeof(Callbacks[0]); i++)
```
Comment (ThFabba) [inline modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: Did this refer to `static PFLT_OPERATION_REGISTRATION Callbacks = NULL;`, i.e. it was using `sizeof(PVOID) / sizeof(FLT_OPERATION_REGISTRATION)` before?

**Comment by ThFabba** [comment]
Could you comment on the lines that were causing warnings please? It's not obvious to me.

File: modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c (hunk 1 above)
Comment (SergeGautherie) [reply modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: Removed.

File: modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c (hunk 2 above)
Comment (SergeGautherie) [reply modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: Removed.

File: modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c [hunk 4]
```diff
@@ -337,7 +339,7 @@ FilterInstanceSetup(
     UNICODE_STRING VolumeName;
     ULONG ReportedSectorSize = 0;
     ULONG SectorSize = 0;
-    NTSTATUS Status;
+    NTSTATUS Status = STATUS_FLT_DO_NOT_ATTACH;
```
Comment (SergeGautherie) [inline modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: {{
c:\reactos-cov\modules\rostests\kmtests\kmtest_drv\kmtest_fsminifilter.c(347,8) :  warning: variable 'Status' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
    if (!(Flags & TESTENTRY_NO_INSTANCE_SETUP))
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
c:\reactos-cov\modules\rostests\kmtests\kmtest_drv\kmtest_fsminifilter.c(398,11) :  note: uninitialized use occurs here
    return Status;
           ^~~~~~
c:\reactos-cov\modules\rostests\kmtests\kmtest_drv\kmtest_fsminifilter.c(347,4) :  note: remove the 'if' if its condition is always true
    if (!(Flags & TESTENTRY_NO_INSTANCE_SETUP))
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
c:\reactos-cov\modules\rostests\kmtests\kmtest_drv\kmtest_fsminifilter.c(340,19) :  note: initialize the variable 'Status' to silence this warning
    NTSTATUS Status;
                   ^
                    = 0
}}

File: modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c (hunk 3 above)
Comment (SergeGautherie) [reply modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: Yes.

{{
c:\reactos-cov\modules\rostests\kmtests\kmtest_drv\kmtest_fsminifilter.c(522,18) :  warning: comparison of unsigned expression < 0 is always false [-Wtautological-unsigned-zero-compare]
    for (i = 0; i < sizeof(Callbacks) / sizeof(Callbacks[0]); i++)
                ~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
}}

File: modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c [hunk 5]
```diff
@@ -543,12 +547,12 @@ FilterPostOperation(
...
     FLT_POSTOP_CALLBACK_STATUS Status;
     UCHAR MajorFunction;
-    INT i;
+    ULONG i;
 
     Status = FLT_POSTOP_FINISHED_PROCESSING;
     MajorFunction = Data->Iopb->MajorFunction;
 
-    for (i = 0; i < sizeof(Callbacks) / sizeof(Callbacks[0]); i++)
+    for (i = 0; i < CallbacksCount; i++)
```
Comment (SergeGautherie) [inline modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: {{
c:\reactos-cov\modules\rostests\kmtests\kmtest_drv\kmtest_fsminifilter.c(551,18) :  warning: comparison of unsigned expression < 0 is always false [-Wtautological-unsigned-zero-compare]
    for (i = 0; i < sizeof(Callbacks) / sizeof(Callbacks[0]); i++)
                ~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
}}

File: modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c [hunk 6]
```diff
@@ -461,7 +463,9 @@ KmtFilterRegisterCallbacks(
         return STATUS_INSUFFICIENT_RESOURCES;
     }
 
-    /* Copy the array, but using the our own pre/post callbacks */
+    CallbacksCount = Count + 1;
```
Comment (ThFabba) [inline modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: You want `Count` here, the last one doesn't have valid data.

File: modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c (hunk 4 above)
Comment (ThFabba) [reply modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: Not sure what the intent is with TESTENTRY_NO_INSTANCE_SETUP, since we have no tests using it yet. Would need Ged's input on whether Don't Attach or Success is better.

File: modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c (hunk 6 above)
Comment (SergeGautherie) [reply modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: Though that changes what existing code seemed to try to do, I agree wrt this PR.
Done. + moved this assignment + added explicit comment + added FIXME comment.

File: modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c (hunk 4 above)
Comment (gedmurphy) [reply modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: Success would be better here. 
The test framework will always register for volume connections, but the caller has the choice to ignore being notified with the TESTENTRY_NO_INSTANCE_SETUP. With this change, if the caller chooses to ignore them, the framework will return STATUS_FLT_DO_NOT_ATTACH, and the filter won't receive any notifications.

File: modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c [hunk 7]
```diff
@@ -470,9 +472,13 @@ KmtFilterRegisterCallbacks(
         Callbacks[i].PostOperation = FilterPostOperation;
     }
 
+    /* FIXME: Useless? See also CallbacksCount value/uses */
```
Comment (gedmurphy) [inline modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: This isn't useless. Windows needs the array to be properly terminated.
What you can't see here is that the array pointer will be set in in the registration struct before the filter is registered with the fltmgr. I just haven't gotten around to that part yet.

Unfortunately, you're submitting patches to unfinished and untested code. I do have some local changes, and will get around to finishing this eventually :/

Comment (SergeGautherie) [reply modules/rostests/kmtests/kmtest_drv/kmtest_fsminifilter.c]: I know this code is unfinished. I assumed this line is intentional. Thanks for the review/explanation.
`FIXME()` removed.
